PortalForecasts.R

```diff
@@ -127,15 +127,17 @@ forecastall <- function(abundances,level,weather_data,weatherforecast) {
                          currency="abundance",model="Forecast", level=level, species="total", estimate=model01$mean, 
                          LowerPI=model01$lower[,which(model01$level==90)], UpperPI=model01$upper[,which(model01$level==90)])
   forecasts01[sapply(forecasts01, is.ts)] <- lapply(forecasts01[sapply(forecasts01, is.ts)],unclass)
-  
+  aic_weights = aic_weights %>%
+    bind_rows(data.frame(date=forecast_date, model='Forecast', currency='abundance', level=level, species='total', aic=model01$model$aic))
   
   model02=forecast(auto.arima(abundances$total,lambda = 0),h=12,level=0.9,fan=T)
   
   forecasts02=data.frame(date=forecast_date, forecastmonth=forecast_months,forecastyear=forecast_years,NewMoonNumber=forecast_newmoons,
                          currency="abundance", model="AutoArima", level=level, species="total", estimate=model02$mean, 
                          LowerPI=model02$lower[,which(model02$level==90)], UpperPI=model02$upper[,which(model02$level==90)])
   forecasts02[sapply(forecasts02, is.ts)] <- lapply(forecasts02[sapply(forecasts02, is.ts)],unclass)
-  
+  aic_weights = aic_weights %>%
+    bind_rows(data.frame(date=forecast_date, model='AutoArima', currency='abundance', level=level, species='total', aic=model02$model$aic))
```

Comment:
If currency, abundance, level, species, and (possibly) date are constant across models, then I'd recommend creating a data.frame with just those values and then making each model's aic_weights data.frame by adding its unique columns on to the template.